### DISCUSSION TITLE
Theme-changer component

### DISCUSSION THREAD

**Discussion by TeamBusylj** [material-components/material-web#5382] [Q&A] [answered]
Hi, is there a planned npm release for the theme-changer component that is on material-web.dev website? I have seen the Typescript reference in the catalog/src directory but i want to use it in javascript. Thanks

**Comment by asyncliz** [comment, marked as answer]
Unfortunately no, there are no plans to provide that as a component. You're welcome to copy the code though!

**Reply by TeamBusylj** [reply]
Thank you for your response! Could you please guide me on where to copy the code from? Should I copy it from here or from the website?
Thanks

**Reply by asyncliz** [reply]
I recommend copying it from the GitHub repo. It's here: https://github.com/material-components/material-web/blob/main/catalog/src/components/theme-changer.ts

**Reply by TeamBusylj** [reply]
Thank you for your response. I'm sorry to bother, but how can i use typescript in browser? I see that other components have js files, but that one doesn't.

**Reply by asyncliz** [reply]
You'll need to use [typescript](https://www.typescriptlang.org/) to compile it into javascript. You can also edit the file and remove the typescript-specific features to turn it into a javascript file.

**Reply by TeamBusylj** [reply]
Thank you so much!
